example/test/react_test_components.dart

```diff
@@ -223,23 +223,168 @@ class _ContextConsumerComponent extends react.Component {
+  }
+}
+
+var newContextRefComponent = react.registerComponent(() => new _NewContextRefComponent());
+
+int calculateChangedBits(currentValue, nextValue) {
+  int result = 1 << 1;
+  if (nextValue['renderCount'] % 2 == 0) {
+    result |= 1 << 2;
+  }
+  return result;
+}
+
+var TestNewContext = createContext({'renderCount': 0}, calculateChangedBits);
+
+class _NewContextProviderComponent extends react.Component2 {
+  _NewContextRefComponent componentRef;
+
+  getInitialState() => {'renderCount': 0, 'complexMap': false};
+
+  printMe() {
+    print('printMe!');
+  }
+
+  render() {
+    Map provideMap = {'renderCount': this.state['renderCount']};
+
+    Map complexValues = {
+      'callback': printMe,
+      'dartComponent': newContextRefComponent,
+      'map': {
+        'bool': true,
+        'anotherDartComponent': newContextRefComponent,
+      },
+      'componentRef': componentRef,
+    };
+
+    if (state['complexMap']) {
+      provideMap.addAll(complexValues);
+    }
+
+    Map newContextRefComponentProps = {
+      'key': 'ref2',
+      'ref': (ref) {
+        componentRef = ref;
+      }
+    };
+
+    return react.ul({
+      'key': 'ulasda',
+    }, [
+      newContextRefComponent(newContextRefComponentProps, []),
+      react.button({
+        'type': 'button',
+        'key': 'button',
+        'className': 'btn btn-primary',
+        'onClick': _onButtonClick,
+      }, 'Redraw'),
+      react.button({
+        'type': 'button',
+        'key': 'button_complex',
+        'className': 'btn btn-primary',
+        'onClick': _onComplexClick,
+      }, 'Redraw With Complex Value'),
+      react.br({'key': 'break1'}),
+      'TestContext.Provider props.value: ${provideMap}',
+      react.br({'key': 'break2'}),
+      react.br({'key': 'break3'}),
+      TestNewContext.Provider(
+        {'key': 'tcp', 'value': provideMap},
+        props['children'],
+      ),
+    ]);
+  }
+
+  _onComplexClick(event) {
+    this.setState({'complexMap': true, 'renderCount': this.state['renderCount'] + 1});
+  }
+
+  _onButtonClick(event) {
+    this.setState({'renderCount': this.state['renderCount'] + 1, 'complexMap': false});
+  }
+}
+
+var newContextProviderComponent = react.registerComponent(() => new _NewContextProviderComponent());
+
+class _NewContextConsumerComponent extends react.Component2 {
+  render() {
+    return TestNewContext.Consumer({'unstable_observedBits': props['unstable_observedBits']}, (value) {
+      return react.ul({
+        'key': 'ul1'
+      }, [
+        'TestContext.Consumer: value = ${value}',
+        react.br({'key': 'break12'}),
+        react.br({'key': 'break22'}),
+        props['children'],
+      ]);
+    });
+  }
+}
+
+var newContextConsumerComponent = react.registerComponent(() => new _NewContextConsumerComponent());
+
+class _NewContextConsumerObservedBitsComponent extends react.Component2 {
+  render() {
+    return TestNewContext.Consumer({'unstable_observedBits': props['unstable_observedBits']}, (value) {
```

Comment:
Same question here.  I don't understand why we would be utilizing an unstable context api... I thought the whole point of this was to transition to the new context _(which surely, is not unstable)_

Reply:
there is a "stable" version of this prop: `observedBits` but it doesn't work as well... Like i mentioned above the whole calculateChangedBits portion isn't even documented by react.